### ISSUE TITLE
[TF2] [Feature Request] Add an option to exclude certain weapons from auto-reloading

### ISSUE THREAD

**Issue by Borloxos** [ValveSoftware/Source-1-Games#4072]
The Force-A-Nature and the Soda Popper eject all cartridges remaining in the magazine when reloading. If auto reload is activated, this means that you lose any rounds you don't shoot immediately unless you switch off the weapon or hold alt-fire until the next shot.
In normal gameplay this is not a big problem, but especially in MvM, where the magazine size of these weapons can be upgraded four times, it can get annoying.

The solution to this problem is to disable automatic reloading in the options, but this means that all other weapons of all other classes will also not automatically reload until the option is re-enabled.
Turning off auto reload in the scout class config isn't great either, as I'd like to keep it on for all other scatterguns and pistols when switching loadouts.

It would be great if there was an option that extends auto reload to exclude either just the Force-A-Nature and Soda-Popper or any list of weapons from auto reload.
Excluding a list would be preferable because it offers more customisation options. For example, there are some complaints about the Ambassador's auto reload obscuring the player's view, but I don't know how severe that has been since the reload animations got altered.

**Comment by AzureAzel** [comment]
I personally think a better option would be just to change those weapons so that they don't dump their entire mag whenever you reload in MvM after upgrading their clip size. It would save the rest of the clip only if you upgraded it's size at least once.

**Comment by Borloxos** [comment]
> I personally think a better option would be just to change those weapons so that they don't dump their entire mag whenever you reload in MvM after upgrading their clip size. It would save the rest of the clip only if you upgraded it's size at least once.

Good idea, but I would also like to have this function for the FaN/Soda Popper in normal matches. Losing a single shell from the clip is not bad, but it still annoys me a lot.
Another way to fix this might be to remove the attribute of losing shells from these two weapons alltogether. I don't think this would affect the balance much.
